source/handbook/javascript-coding-guidelines.rst

```diff
@@ -0,0 +1,67 @@
+############################
+JavaScript Coding Guidelines
+############################
+
+Most of our javascript guidelines are enforced by various linters.
+We are currently using ``eslint``.
+We encourage you to familiarize yourself with the settings.
+
+NodeJS Version
+**************
+
+We generally strive to support the newest Active LTS NodeJS version.
+
+File / Directory Layout
+***********************
+
+If a javascript file contains a single class, the file should be named after the class and be UpperCamelCase.
+If a javascript file contains utility functions, the file should be given a descriptive name and be snake_case.
+
+We expect parallel source and test directory hierarchies.
+For example,
+
+::
+
+    root
+      - parser  # package name, directory containing source code
+        Parser.js  # contains class Parser
+        - ast
+          - ast.js  # contains various exports
+
+      - tests  # directory containing tests for source code
+        - Parser_spec.js  # unit tests for parser/Parser.js
+        - ast
+          - ast_spec.js  # unit tests for parser/ast/ast.js
+
+      - package.json
+
+Naming
+******
+
```

Comment:
Should we change to:

1. camelCase should be used for variables, methods and functions. Private variables should be prepended with an underscore. 
2. PascalCase should be used for class and component.
3. TITLE_CASE should be used for constants.

Reply:
yes, as discussed in Discord this needs updating